src/sync/sync_manager.hpp

```diff
@@ -49,22 +52,70 @@ class SyncLoggerFactory {
 };
 
 class SyncManager {
-friend struct SyncSession;
+friend class SyncSession;
 public:
+    enum class MetadataMode {
+        NoEncryption,                   // Enable metadata, but disable encryption.
+        Encryption,                     // Enable metadata, and use encryption (automatic if possible).
+        NoMetadata,                     // Disable metadata.
+    };
+
     static SyncManager& shared();
 
+    // Configure the metadata and file management subsystems. This MUST be called upon startup.
+    void configure_file_system(const std::string& base_file_path,
+                               MetadataMode metadata_mode=MetadataMode::Encryption,
+                               util::Optional<std::vector<char>> custom_encryption_key=none);
+
     void set_log_level(util::Logger::Level) noexcept;
     void set_logger_factory(SyncLoggerFactory&) noexcept;
     void set_error_handler(std::function<sync::Client::ErrorHandler>);
 
     /// Control whether the sync client attempts to reconnect immediately. Only set this to `true` for testing purposes.
     void set_client_should_reconnect_immediately(bool reconnect_immediately);
+    bool client_should_reconnect_immediately() const;
+
     /// Control whether the sync client validates SSL certificates. Should *always* be `true` in production use.
-    void set_client_should_validate_ssl(bool validate_ssl);
+    void set_client_should_validate_ssl(bool validate_ssl)
+    {
+        std::lock_guard<std::mutex> lock(m_mutex);
+        m_client_validate_ssl = validate_ssl;
+    }
+
+    bool client_should_validate_ssl() const {
+        std::lock_guard<std::mutex> lock(m_mutex);
+        return m_client_validate_ssl;
+    }
 
     std::shared_ptr<SyncSession> get_session(const std::string& path, const SyncConfig& config);
     std::shared_ptr<SyncSession> get_existing_active_session(const std::string& path) const;
 
+    util::Logger::Level log_level() const noexcept {
+        std::lock_guard<std::mutex> lock(m_mutex);
+        return m_log_level;
+    }
```

Comment:
Note that locking the `std::mutex` is not `noexcept`, so this risks terminating the program in case there is a mutex error (which may be the reasonable thing to do).